bindings/matrix-sdk-ffi/src/authentication_service.rs

```diff
@@ -55,10 +93,120 @@ impl From<IdParseError> for AuthenticationError {
     }
 }
 
+impl From<OidcError> for AuthenticationError {
+    fn from(e: OidcError) -> AuthenticationError {
+        AuthenticationError::OidcError { message: e.to_string() }
+    }
+}
+
+/// The configuration to use when authenticating with OIDC.
+#[derive(uniffi::Record)]
+pub struct OidcConfiguration {
+    /// The name of the client that will be shown during OIDC authentication.
+    pub client_name: String,
+    /// The redirect URI that will be used when OIDC authentication is
+    /// successful.
+    pub redirect_uri: String,
+    /// A URI that contains information about the client.
+    pub client_uri: String,
+    /// A URI that contains the client's terms of service.
+    pub tos_uri: String,
+    /// A URI that contains the client's privacy policy.
+    pub policy_uri: String,
+
+    /// Pre-configured registrations for use with issuers that don't support
+    /// dynamic client registration.
+    pub static_registrations: HashMap<String, String>,
+}
+
+/// The data needed to restore an OpenID Connect session.
+#[derive(Debug, Serialize, Deserialize)]
+struct OidcRegistrations {
```

Comment:
Also the same here, @hughns do you think we need to encrypt the Client ID?

Note on using the state store - we don't have one at this stage as EX creates a directory per user like so: `base_path/user_id`. So this data is being stored in `base_path/oidc/registrations.json` so it can be accessed across all users.

Reply:
> Note on using the state store - we don't have one at this stage as EX creates a directory per user like so: base_path/user_id. So this data is being stored in base_path/oidc/registrations.json so it can be accessed across all users.

Ah right I forgot about that :sweat_smile:

Reply:
> Also the same here, @hughns do you think we need to encrypt the Client ID?

My understanding is that the `client_id` is public data as far as OIDC is concerned and so does not need to be encrypted at rest.

Reply:
Thanks, then lets leave this as a JSON file.